Doc/library/tkinter.rst

```diff
@@ -11,100 +11,267 @@
 --------------
 
 The :mod:`tkinter` package ("Tk interface") is the standard Python interface to
-the Tk GUI toolkit.  Both Tk and :mod:`tkinter` are available on most Unix
-platforms, as well as on Windows systems.  (Tk itself is not part of Python; it
-is maintained at ActiveState.)
+the Tcl/Tk GUI toolkit. Tcl/Tk and :mod:`tkinter` are available on most Unix
+platforms (including macOS), as well as on Windows systems.
 
 Running ``python -m tkinter`` from the command line should open a window
 demonstrating a simple Tk interface, letting you know that :mod:`tkinter` is
 properly installed on your system, and also showing what version of Tcl/Tk is
 installed, so you can read the Tcl/Tk documentation specific to that version.
 
-.. seealso::
-
-   Tkinter documentation:
-
-   `Python Tkinter Resources <https://wiki.python.org/moin/TkInter>`_
-      The Python Tkinter Topic Guide provides a great deal of information on using Tk
-      from Python and links to other sources of information on Tk.
-
-   `TKDocs <http://www.tkdocs.com/>`_
-      Extensive tutorial plus friendlier widget pages for some of the widgets.
-
-   `Tkinter reference: a GUI for Python <https://infohost.nmt.edu/tcc/help/pubs/tkinter/web/index.html>`_
-      On-line reference material.
-
-   `Tkinter docs from effbot <http://effbot.org/tkinterbook/>`_
-      Online reference for tkinter supported by effbot.org.
-
-   `Programming Python <http://learning-python.com/about-pp4e.html>`_
-      Book by Mark Lutz, has excellent coverage of Tkinter.
-
-   `Modern Tkinter for Busy Python Developers <https://www.amazon.com/Modern-Tkinter-Python-Developers-ebook/dp/B0071QDNLO/>`_
-      Book by Mark Rozerman about building attractive and modern graphical user interfaces with Python and Tkinter.
-
-   `Python and Tkinter Programming <https://www.manning.com/books/python-and-tkinter-programming>`_
-      Book by John Grayson (ISBN 1-884777-81-3).
-
-   Tcl/Tk documentation:
+Tkinter supports a range of Tcl/Tk versions, built either with or
+without thread support. The official Python binary release bundles Tcl/Tk 8.6
+threaded. See the source code for the :mode:`_tkinter` module
+for more information about supported versions.
+
+Tkinter is not a thin wrapper, but adds a fair amount of its own logic to
+make the experience more pythonic. This documentation will concentrate on these
+additions and changes, and refer to the official Tcl/Tk documentation for
+descriptions of underlying features.
+
+
+Architecture
+------------
+
+Tkinter provides a unified interface to several separate and distinct libraries.
```

Comment:
It's not necessarily a bundle. The source versions are released as two separate tarballs, from two separate repositories. Many Linux distros distribute them as separate rpm's, meaning sometimes you get Tcl installed on a system but not Tk (usually when Tcl is pulled in as a dependency of some other package). Surprised me the first time I ran into it actually. Other times, e.g. with ActiveState, it is a single bundle. Your point is worth addressing, but I'm going to suggest doing it a bit later when it mentions the Tk library.

As a historical point of interest, distributing it as a strict core plus extensions available separately was another one of Ousterhout's "things", though many of us were pushing for a "batteries included" approach as Python uses. 

I also fail to see why you refer to this as "meaningless" as an intro sentence to an architecture section. But whatever.

Reply:
I'm fine with what's here.

Reply:
It doesn't matter how they happen to be packaged. From architecture and documentation POV, they are separate.
I don't really see a harm in this design: last I checked, the separation of concerns principle still existed. It's just that interested parties need to know this fact, that's all.
The fact must be in the header of "Architecture" section because it justifies the section's existence. For most other bindings, just describing the interface is enough.
"several separate and distinct libraries" doesn't mean anything and only confuses the reader: "they said it's an interface to Tcl/Tk -- and now, it's some unspecified "several separate libraries"? The 1st phrase dulpicates the intro. So, the sentence has negative informativity.

Reply:
https://github.com/roseman/cpython/pull/3